neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,304 @@
+            # 'd' means digital. Per discussion with neuroconv users the use of
+            # 'a.u.' makes the units clearer
+            elif channel_info["chan_type"][channel_index][0] == "d":
+                units = "a.u."
+            # aux channel
+            else:
+                units = "V"
+
+            signal_channels.append(
+                (
+                    channel_name,
+                    channel_id,
+                    self._sampling_frequency,
+                    BINARY_DTYPE,
+                    units,
+                    1,  # no gain
+                    0,  # no offset
+                    stream_id,
+                )
+            )
+
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+
+        stream_ids = np.unique(signal_channels["stream_id"])
+        signal_streams = np.zeros(stream_ids.size, dtype=_signal_stream_dtype)
+        signal_streams["id"] = [str(stream_id) for stream_id in stream_ids]
+        for stream_index, stream_id in enumerate(stream_ids):
+            name = stream_id_to_stream_name.get(int(stream_id), "")
+            signal_streams["name"][stream_index] = name
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+
+        # Put all the necessary info in the header
+        self.header = {}
+        self.header["nb_block"] = 1
+        self.header["nb_segment"] = [1]
+        self.header["signal_streams"] = signal_streams
+        self.header["signal_channels"] = signal_channels
+        self.header["spike_channels"] = spike_channels
+        self.header["event_channels"] = event_channels
+
+        # Add the minimum annotations
+        self._generate_minimal_annotations()
+
+        # date comes out as:
+        # year-month-daydayofweektime all as a string so we need to prep it for
+        # entering into datetime
+        # example: '2024-07-01T13:04:49.4972245-04:00'
+        stringified_date_list = self.metadata['status']['start_time'].split('-')
+        year = int(stringified_date_list[0])
+        month = int(stringified_date_list[1])
+        day = int(stringified_date_list[2][:2]) # day should be first two digits of the third item in list
+        time_info = stringified_date_list[2].split(':')
+        hour = int(time_info[0][-2:])
+        minute = int(time_info[1])
+        second = int(time_info[2])
+        microsecond = 1000 * 1000 * (float(time_info[2]) - second) # second -> micro is 1000 * 1000
+
+        rec_datetime = datetime.datetime(year, month, day, hour, minute, second, microsecond)
+        bl_annotations = self.raw_annotations["blocks"][0]
+        seg_annotations = bl_annotations["segments"][0]
+        for d in (bl_annotations, seg_annotations):
+            d["rec_datetime"] = rec_datetime
+
```

Comment:
@samuelgarcia so I don't do as much date time stuff you want to look at this. Basically we get this string. I don't know what "T" inside the string or the 4:00 at the end (maybe duration in seconds?) so I did some ugly parsing. If you have a better idea from the parsing let me know!

Reply:
I guess my fear with this type of parsing is that if they change how they encode it we end up with giant parsing logic.

Reply:
2024-07-01T13:04:49.4972245-04:00
date_time_{T}_{timestamps}_{time_zone}

Reply:
@h-mayorquin if you have a good system could you actually make a PR off of this one and I'll merge it into this. I'm trash at this parsing and it seems like you have a better idea of how to pull this out!

Reply:
We are ready to merge this @h-mayorquin  unless you have a better idea for parsing. Do you want me to hold off until you take a look or good to merge and we can prettify this bit later?